app/src/main/java/com/codepath/apps/restclienttemplate/ComposeDialogFragment.java

```diff
@@ -0,0 +1,148 @@
+package com.codepath.apps.restclienttemplate;
+
+import android.content.Intent;
+import android.media.Image;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.KeyEvent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.view.inputmethod.EditorInfo;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import com.codepath.apps.restclienttemplate.models.Tweet;
+import com.codepath.asynchttpclient.callback.JsonHttpResponseHandler;
+
+import org.json.JSONException;
+import org.parceler.Parcels;
+
+import okhttp3.Headers;
+
+
+public class ComposeDialogFragment extends DialogFragment implements TextView.OnEditorActionListener {
+
+    public static final String TAG = "ComposeDialogFragment";
+    public static final int MAX_TWEET_LENGTH = 280;
+
+    EditText etCompose;
+    Button btnTweet;
+    ImageView ivClose;
+    TwitterClient client;
+
+    Tweet result;
+
+    public ComposeDialogFragment() {
+        // Empty constructor is required for DialogFragment
+        // Make sure not to add arguments to the constructor
+        // Use `newInstance` instead as shown below
+    }
+
+    public interface ComposeDialogListener {
+        void onFinishComposeDialog(Tweet tweet);
+    }
+
+    public static ComposeDialogFragment newInstance(String replyTo) {
+        ComposeDialogFragment frag = new ComposeDialogFragment();
+        Bundle args = new Bundle();
+        args.putString("in_reply_to", replyTo);
```

Comment:
I'd prefer a final string REPLY_TO_KEY = "in_reply_to" so that you can easily use the key when retrieving this information without remember or look up this key